pyomo/contrib/fme/fourier_motzkin_elimination.py

```diff
@@ -301,56 +394,144 @@ def _fourier_motzkin_elimination(self, constraints, vars_to_eliminate):
             geq_list = []
             waiting_list = []
 
+            coefs = []
             for cons in constraints:
                 leaving_var_coef = cons['map'].get(the_var)
                 if leaving_var_coef is None or leaving_var_coef == 0:
                     waiting_list.append(cons)
+                    if self.verbose:
+                        logger.info("\t%s <= %s" 
+                                    % (cons['lower'], 
+                                       cons['body'].to_expression()))
                     continue
 
                 # we know the constraint is a >= constraint, using that
                 # assumption below.
                 # NOTE: neither of the scalar multiplications below flip the
                 # constraint. So we are sure to have only geq constraints
                 # forever, which is exactly what we want.
-                if leaving_var_coef < 0:
-                    leq_list.append(
-                        self._nonneg_scalar_multiply_linear_constraint(
-                            cons, -1.0/leaving_var_coef))
+                if not self.do_integer_arithmetic:
+                    if leaving_var_coef < 0:
+                        leq_list.append(
+                            self._nonneg_scalar_multiply_linear_constraint(
+                                cons, -1.0/leaving_var_coef))
+                    else:
+                        geq_list.append(
+                            self._nonneg_scalar_multiply_linear_constraint(
+                                cons, 1.0/leaving_var_coef))
                 else:
-                    geq_list.append(
-                        self._nonneg_scalar_multiply_linear_constraint(
-                            cons, 1.0/leaving_var_coef))
+                    if not _is_integer(leaving_var_coef, self.integer_tolerance):
+                        raise ValueError(self._get_noninteger_coef_error_message(
+                            the_var.name, leaving_var_coef))
+                    coefs.append(int(leaving_var_coef))
```

Comment:
This is dangerous.  Consider the case:
```python
leaving_var_coef = 0.999
self.integer_tolerance = 0.01
_is_integer(leaving_var_coef, self.integer_tolerance) # == True!
int(leaving_var_coef)  # 0!
```

The preferred way to round is `int(leaving_var_coef + 0.5)`